File: backend/src/main/java/com/romeo/VetLink/vetClinic/domain/VetClinic.java
package com.romeo.VetLink.vetClinic.domain;

import com.romeo.VetLink.config.OwnedEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.*;

@EqualsAndHashCode(callSuper = true)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "vet_clinic")
public class VetClinic extends OwnedEntity {
    @Id
    @GeneratedValue
    private Integer id;

    private String name;
    private String county;
    private String city;
    private String street;
    private String number;
    private String phoneNumber;
    private String shortDescription;


}

File: mobile/lib/model/vetClinic.dart
class VetClinic {
  final int id;
  final String name;
  final String county;
  final String city;
  final String street;
  final String number;
  final String phoneNumber;
  final String shortDescription;
  final String owner;

  const VetClinic({
    required this.id,
    required this.name,
    required this.county,
    required this.city,
    required this.street,
    required this.number,
    required this.phoneNumber,
    required this.shortDescription,
    required this.owner,
  });


  static VetClinic fromMap(Map<String, dynamic> map) {
    return VetClinic(
      id: map['id'],
      name: map['name'],
      county: map['county'],
      city: map['city'],
      street: map['street'],
      number: map['number'],
      phoneNumber: map['phoneNumber'],
      shortDescription: map['shortDescription'],
      owner: map['owner'],
    );
  }


  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'county': county,
      'city': city,
      'street': street,
      'number': number,
      'phoneNumber': phoneNumber,
      'shortDescription': shortDescription,
      'owner': owner,
    };
  }
}
